src/auxinfo/proof.rs

```diff
@@ -143,3 +143,193 @@ impl AuxInfoProof {
+        ParticipantIdentifier,
+    };
+    use rand::random;
+
+    use super::*;
+
+    pub(crate) fn generate_shared_context() -> SharedContext {
+        let mut rng = init_testing();
+        let sid = Identifier::random(&mut rng);
+        let participant = ParticipantIdentifier::random(&mut rng);
+        let participant2 = ParticipantIdentifier::random(&mut rng);
+        let participants = vec![participant, participant2];
+        let generator = CurvePoint::GENERATOR;
+        let order = k256_order();
+        SharedContext::gen_shared_context(sid, participants, generator, order)
+    }
+
+    #[test]
+    fn auxinfo_proof_verifies() -> Result<()> {
+        let mut rng = init_testing();
+        let sid = Identifier::random(&mut rng);
+        let rho = random();
+        let setup_params = VerifiedRingPedersen::gen(&mut rng, &())?;
+        let (p, q) = prime_gen::get_prime_pair_from_pool_insecure(&mut rng).unwrap();
+        let modulus = &p * &q;
+        let shared_context = &generate_shared_context();
+        let proof = AuxInfoProof::prove(
+            &mut rng,
+            shared_context,
+            sid,
+            rho,
+            &setup_params,
+            &modulus,
+            &p,
+            &q,
+        )?;
+        assert!(proof
+            .verify(shared_context, sid, rho, &setup_params, &modulus)
+            .is_ok());
+        Ok(())
+    }
+
+    #[test]
+    fn each_constituent_proof_must_be_valid() -> Result<()> {
+        let mut rng = init_testing();
+        let sid = Identifier::random(&mut rng);
+        let rho = random();
+        let setup_params = VerifiedRingPedersen::gen(&mut rng, &())?;
+        let (p, q) = prime_gen::get_prime_pair_from_pool_insecure(&mut rng).unwrap();
+        let modulus = &p * &q;
+        let shared_context = &generate_shared_context();
+        let mut transcript = AuxInfoProof::new_transcript();
+        AuxInfoProof::append_pimod_transcript(&mut transcript, shared_context, sid, rho)?;
+        let pimod = pimod::PiModProof::prove(
+            &pimod::CommonInput::new(&modulus),
+            &pimod::ProverSecret::new(&p, &q),
+            shared_context,
+            &mut transcript,
+            &mut rng,
+        )?;
+        AuxInfoProof::append_pifac_transcript(&mut transcript, shared_context, sid, rho)?;
+        let pifac = pifac::PiFacProof::prove(
+            pifac::CommonInput::new(&setup_params, &modulus),
+            pifac::ProverSecret::new(&p, &q),
+            shared_context,
+            &mut transcript,
+            &mut rng,
+        )?;
+        let proof = AuxInfoProof {
+            pimod: pimod.clone(),
+            pifac: pifac.clone(),
+        };
+        assert!(proof
+            .verify(shared_context, sid, rho, &setup_params, &modulus)
+            .is_ok());
+        let mut rng1 = init_testing();
+        let sid1 = Identifier::random(&mut rng1);
+        let rho1: [u8; 32] = random();
```

Comment:
todo: use `rng.gen()` instead of this method. This internally creates a `thread_rng` that is not connected to our RNG, so it makes the tests non-replicable if they fail.

Reply:
Please remove the `random()` method throughout the tests, there are a couple of other instances.